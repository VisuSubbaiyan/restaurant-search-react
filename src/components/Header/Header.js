import React from 'react';
import PropTypes from 'prop-types';

import './Header.scss';

const Header = props => (
  <header className="header">
    <h1 className="heading">{props.heading}</h1>
  </header>
);

Header.propTypes = {
  heading: PropTypes.string.isRequired
}


export default React.memo(Header);
