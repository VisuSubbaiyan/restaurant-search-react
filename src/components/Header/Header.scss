.header {
  background: var(--grey-dark);
  padding: 10px;
  color: var(--grey-light);
  display: flex;
  align-items: center;
  position: fixed;
  width: 100%;
}

.header h1.heading {
  margin: 0;
  padding: 10px 16px;
}