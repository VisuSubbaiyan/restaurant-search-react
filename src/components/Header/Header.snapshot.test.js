import React from 'react';

import Header from './Header';
import { createMountedComponent } from '../../utils/testUtils';

describe('<Header /> should', () => {
  const mountedComponent = createMountedComponent({
    children: <Header heading="sample Text" />
  });

  const heading = mountedComponent.find('h1');

  it('render correctly according to props', () => {
    expect(mountedComponent.htmlRender()).toMatchSnapshot();
    expect(heading).toHaveLength(1);
  })
});