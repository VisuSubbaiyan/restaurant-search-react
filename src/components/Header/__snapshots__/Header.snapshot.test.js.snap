// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should render correctly according to props 1`] = `
"<header class=\\"header\\">
  <h1 class=\\"heading\\">sample Text</h1>
</header>"
`;
